### PR TITLE
feat: make the paginator key bindings customizable

### LINKED ISSUE #271: paginator: does not support custom key bindings
Following the example of #269 - I believe this may be the last bubble missing key bindings.

I'll send a PR.

### PR DESCRIPTION
**Pull request by knz**
Fixes #271.
This will also make the paginator benefit from the improvement in #261.

cc @maaslalani @meowgorithm

### CODE REVIEW

File: paginator/paginator.go [hunk 1]
```diff
@@ -19,20 +20,49 @@ const (
...
-	UseHLKeys         bool
-	UseJKKeys         bool
+	// UseLeftRightKeys is deprecated. Customize KeyMap instead.
+	UseLeftRightKeys bool
+	// UseUpDownKeys is deprecated. Customize KeyMap instead.
+	UseUpDownKeys bool
+	// UseHLKeys is deprecated. Customize KeyMap instead.
+	UseHLKeys bool
+	// UseJKKeys is deprecated. Customize KeyMap instead.
+	UseJKKeys bool
```
Comment (maaslalani) [inline paginator/paginator.go:65]: From https://go.dev/blog/godoc

> Sometimes a struct field, function, type, or even a whole package becomes redundant or unnecessary, but must be kept for compatibility with existing programs. To signal that an identifier should not be used, add a paragraph to its doc comment that begins with “Deprecated:” followed by some information about the deprecation.

**Comment by maaslalani** [comment]
This looks good to me behaviour wise since I did a quick code search on GitHub repos using the `Use**Keys` API and mostly found Charm repos and gum forks along with one other repo using it https://github.com/bensadeh/circumflex/blob/0d2b5b24cf9a69eb0eee800dd0f57e3ab31cf503/bubble/list/list.go#L152

image

File: paginator/paginator.go (hunk 1 above)
Comment (maaslalani) [inline paginator/paginator.go:65]:
```suggestion
	// Deprecated: customize KeyMap instead.
	UseLeftRightKeys bool
	// Deprecated: customize KeyMap instead.
	UseUpDownKeys bool
	// Deprecated: customize KeyMap instead.
	UseHLKeys bool
	// Deprecated: customize KeyMap instead.
	UseJKKeys bool
```

File: paginator/paginator.go [hunk 2]
```diff
@@ -19,20 +20,49 @@ const (
...
+	ActiveDot string
+	// InactiveDot is used to mark inactive pages under the Dots display type.
+	InactiveDot string
+	// ArabicFormat is the printf-style format to use for the Arabic display type.
+	ArabicFormat string
+
+	// KeyMap encodes the keybindings recognized by the widget.
+	KeyMap KeyMap
+
+	// UsePgUpPgDownKeys is deprecated. Customize KeyMap instead.
```
Comment (maaslalani) [inline paginator/paginator.go:56]:
```suggestion
	// Deprecated: customize KeyMap instead.
```

**Review by maaslalani — APPROVED** [review]

File: paginator/paginator.go (hunk 1 above)
Comment (knz) [reply paginator/paginator.go]: Done.

File: paginator/paginator.go (hunk 2 above)
Comment (knz) [reply paginator/paginator.go]: Done.

File: paginator/paginator.go [hunk 3]
```diff
@@ -19,20 +20,49 @@ const (
...
-	Type              Type
-	Page              int
-	PerPage           int
-	TotalPages        int
-	ActiveDot         string
-	InactiveDot       string
-	ArabicFormat      string
+	// Type configures how the pagination is rendered (Arabic, Dots).
+	Type Type
+	// Pag is the current page number.
```
Comment (maaslalani) [inline paginator/paginator.go:40]:
```suggestion
	// Page is the current page number.
```

**Review by maaslalani — APPROVED** [review]

File: paginator/paginator.go (hunk 3 above)
Comment (knz) [reply paginator/paginator.go]: Done.

**Comment by maaslalani** [comment]
Awesome work @knz, thank you!

**Pull request merged**